riscv-page-tables/src/page_table.rs

```diff
@@ -660,29 +659,38 @@ impl<T: PagingMode> PageTableInner<T> {
             .checked_add_pages_with_size(num_pages, page_size)
             .ok_or(Error::OutOfMapRange)?;
 
-        let iterator = start_addr
-            .iter_from_with_size(page_size)
-            .ok_or_else(|| Error::AddressMisaligned(start_addr.bits()))?;
+        let mut remaining = num_pages;
+        let mut vaddr = start_addr;
 
-        for vaddr in iterator.take(num_pages as usize) {
+        while remaining > 0 {
             let mut table = PageTable::from_root(self);
```

Comment:
You could move this before the loop:
```rust
let root_table = PageTable::from_root(self);
while remaining > 0 {
        let mut table = root_table;
```

Reply:
Ah, now I remember why. :-)

The value is considered moved once assigned to table. So has to be recreated every time anyway.

Reply:
Ah too bad :)